lib/assessment.js

```diff
@@ -29,7 +46,11 @@ var newArray = function newArray(length, defaultValue) {
 // invoking the argument defaultsFunction
 // defaultsFunction should be passed the index of the array element to be set
 var newArray2 = function newArray2(length, defaultsFunction) {
-
+  for (var i = 0; i < length; i++) {
+    newArray2 [i] = defaultsFunction();
+    // newArray2 [i] =
+  }
+  return newArray2;
 };
```

Comment:
Here, we see a similar problem to above, in that you did not declare a new variable, create a new array, and assign the array to the variable. Imagine this is the first line in your function's body, before the `for` loop: `var arrayToReturn = [];`

Without an array to work with, and a for loop to put a statement inside, anyone would be confused about what to do on line 50. What your line 50 does is assign the result of invoking `defaultsFunction` without any arguments to a numeric property on the function that is currently executing. Finally, the function returns itself.

The reason I describe this as similar to your answer to question 4 is because it returns the same reference every time you run it. If you were to invoke this function several times, the function object (stored in `newArray2`) will have attributes added to it and overwritten in subsequent invocations, and the return value would always be the function itself. Anything accessing the attributes of the returned function will be working with an unsafe store of data that can be overwritten without notice at any time.

1/4 credit.